scripts/bulk-mailer/normalize-user-records.js

```diff
@@ -0,0 +1,92 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict'
+
+const leftpad = require('leftpad')
+
+module.exports = class UserRecordNormalizer {
+  /**
+   * Normalize the records in `userRecords`. Filters out entries
+   * without an `email` field, converts location information to
+   * the a format suitable for the user.
+   *
+   * @param {Object[]} userRecords
+   * @param {Object} translator
+   */
+  normalize(userRecords, translator) {
+    return userRecords
+      // no email can be sent if the record does not contain an email
+      .filter(record => !! record.email)
+      .map(userRecord => this.normalizeUserRecord(userRecord, translator))
+  }
+
+  normalizeUserRecord (userRecord, translator) {
+    this.normalizeAcceptLanguage(userRecord)
+    this.normalizeLanguage(userRecord, translator)
+    this.normalizeLocations(userRecord)
+
+    return userRecord
+  }
+
+  normalizeAcceptLanguage(userRecord) {
+    // The Chinese translations were handed to us as "zh" w/o a country
```

Comment:
For a user in stage that signed up with `accept-language: fr-FR`, when I send a bulk email to them, I'm not getting a translation in some types of email of interest. Is that expected?

Reply:
This should now be fixed.